src/enclave/inside/env.cpp

```diff
@@ -1,21 +1,42 @@
+#include <enclave/inside/EnclaveWasmModule.h>
 #include <enclave/inside/native.h>
+#include <enclave/inside/ocalls.h>
+
+#include <string>
 
 namespace sgx {
-static int wasi_args_get(wasm_exec_env_t exec_env, int a, int b)
+static int wasi_args_get(wasm_exec_env_t exec_env,
+                         uint32_t* argvOffsetsWasm,
+                         char* argvBuffWasm)
 {
+    ocallLogDebug("S - wasi_args_get");
+
+    GET_EXECUTING_MODULE_AND_CHECK(exec_env);
+
+    module->writeArgvToWamrMemory(argvOffsetsWasm, argvBuffWasm);
+
     return 0;
 }
 
-static int wasi_args_sizes_get(wasm_exec_env_t exec_env, int a, int b)
+static int wasi_args_sizes_get(wasm_exec_env_t exec_env,
+                               uint32_t* argcWasm,
+                               uint32_t* argvBuffSizeWasm)
 {
+    ocallLogDebug("S - wasi_args_sizes_get");
+
+    GET_EXECUTING_MODULE_AND_CHECK(exec_env);
+
+    *argcWasm = module->getArgc();
+    *argvBuffSizeWasm = module->getArgvBufferSize();
```

Comment:
I think native symbols aren't such good candidates to be included in a mixin. The symbol registration doesn't trivially work well with templated functions, and there are some methods (like logging and getting the executing module) that have different names and signatures.

That being said, after the new mixin structure the methods body is either trivial or implemented in the `WAMRModuleMixin`.

Reply:
I don't mean we should merge the whole native symbol thing in, just the logic of the body of the function. This is how we share host interface logic between WAVM and WAMR normally, e.g. having a method like `module->setUpArgcArgv(argcWasm, argvBuffSizeWasm)` rather individual calls to `getArgc` and `getArgvBufferSize`. Admittedly here it will only save a couple of lines, but it's more the principle of absolutely minimising duplication.